engine/crates/engine-v2/schema/src/lib.rs

```diff
@@ -269,3 +269,12 @@ impl ScalarType {
         })
     }
 }
+
+impl Graph {
+    pub fn scalar_definition_by_name(&self, name: &str) -> Option<ScalarDefinitionId> {
+        self.scalar_definitions
+            .iter()
+            .position(|definition| definition.ty.to_string() == name)
+            .map(ScalarDefinitionId::from)
+    }
+}
```

Comment:
We could do either a lookup map for these in the graph, or we could just mark the ID of boolean when generating the schema?

Reply:
And could this return `DefinitionId` already?

Reply:
For the latter sure, for the former this was a suggestion from @Finistere so there might be some reason not to do so, but I agree in general. @Finistere WDYT?

We could probably do this separately though to avoid delaying

Reply:
Fixed the definition ID issue, we'll tackle creating a lookup map separately